### DISCUSSION TITLE
/bin/sh: java: command not found

### DISCUSSION THREAD

**Discussion by zhuli20** [apache/dolphinscheduler#11707] [Q&A]
[LOG-PATH]: /home/dolphin/dolphin-bin/worker-server/logs/20220831/6631814426144_1-6-14.log, [HOST]:  Host{address='10.15.135.138:1234', ip='10.15.135.138', port=1234}
[INFO] 2022-08-31 03:07:10.761 +0000 [taskAppId=TASK-20220831-6631814426144_1-6-14] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.datax.DataxTask:[134] - datax task params {"localParams":[],"resourceList":[],"customConfig":0,"dsType":"MYSQL","dataSource":"2","dtType":"POSTGRESQL","dataTarget":"1","sql":"select name as sales_unit,position as indicator_name,1 as alert_value from uf_bi_user_re limit 10","targetTable":"public.test","jobSpeedByte":0,"jobSpeedRecord":0,"preStatements":[],"postStatements":[],"xms":1,"xmx":1}
[INFO] 2022-08-31 03:07:10.769 +0000 [taskAppId=TASK-20220831-6631814426144_1-6-14] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.datax.DataxTask:[85] - tenantCode user:dolphin, task dir:6_14
[INFO] 2022-08-31 03:07:10.770 +0000 [taskAppId=TASK-20220831-6631814426144_1-6-14] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.datax.DataxTask:[90] - create command file:/tmp/dolphinscheduler/exec/process/6631176595616/6631814426144_1/6/14/6_14.command
[INFO] 2022-08-31 03:07:10.771 +0000 [taskAppId=TASK-20220831-6631814426144_1-6-14] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.datax.DataxTask:[116] - command : #!/bin/sh
BASEDIR=$(cd `dirname $0`; pwd)
cd $BASEDIR
source /home/dolphin/dolphin-bin/worker-server/conf/dolphinscheduler_env.sh
/tmp/dolphinscheduler/exec/process/6631176595616/6631814426144_1/6/14/6_14_node.sh
[INFO] 2022-08-31 03:07:10.773 +0000 [taskAppId=TASK-20220831-6631814426144_1-6-14] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.datax.DataxTask:[290] - task run command: sudo -u dolphin sh /tmp/dolphinscheduler/exec/process/6631176595616/6631814426144_1/6/14/6_14.command
[INFO] 2022-08-31 03:07:10.775 +0000 [taskAppId=TASK-20220831-6631814426144_1-6-14] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.datax.DataxTask:[181] - process start, process id is: 24028
[INFO] 2022-08-31 03:07:10.865 +0000 [taskAppId=TASK-20220831-6631814426144_1-6-14] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.datax.DataxTask:[205] - process has exited, execute path:/tmp/dolphinscheduler/exec/process/6631176595616/6631814426144_1/6/14, processId:24028 ,exitStatusCode:127 ,processWaitForStatus:true ,processExitValue:127
[INFO] 2022-08-31 03:07:11.776 +0000 [taskAppId=TASK-20220831-6631814426144_1-6-14] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.datax.DataxTask:[63] -  -> 
	DataX (DATAX-OPENSOURCE-3.0), From Alibaba !
	Copyright (C) 2010-2017, Alibaba Group. All Rights Reserved.
	
	/bin/sh: java: command not found
[INFO] 2022-08-31 03:07:11.778 +0000 [taskAppId=TASK-20220831-6631814426144_1-6-14] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.datax.DataxTask:[57] - FINALIZE_SESSION

**Comment by DarkAssassinator** [comment]
u should set the environment. or create Soft link jdk to /usr/bin/java

**Comment by zhuyt1090** [comment]
I have also encountered this problem. The solution is to modify the datax.py configuration file

![image](https://user-images.githubusercontent.com/108874436/188412201-99905709-5966-4e49-82ca-b9b4fc968b1d.png)
